api.ts

```diff
@@ -1282,12 +1282,50 @@ export type MetadataFormula = BaseFormula<[ParamDef<Type.string>, ParamDef<Type.
   schema?: any;
 };
 
-type PropertyAutocompleteMetadataFormula = BaseFormula<[], any> & {
+type GenericMetadataFormula = BaseFormula<ParamDefs, any> & {
+  schema?: any;
+};
+
+/**
+ * These will be created with a helper function.
+ */
+interface PropertyAutocompleteFormattedResult {
+  __brand: 'PropertyAutocompleteSimpleResult';
+  /** Text that will be displayed to the user in UI for this option. */
+  display: string;
+  /** The actual value for this option */
+  value: any;
+}
+
+type PropertyAutocompleteResults =
+  | any[]
+  | {
+      cacheTtlSecs?: number;
+      results: Array<any | PropertyAutocompleteFormattedResult>;
```

Comment:
Do we need the `any |` here? If you're going to return a formatted result object, might as well require that the individual results are also all formatted too?

Reply:
And similar here, if the pack returns scalars the SDK should convert them to PropertyAutocompleteFormattedResult and then we should get a strict type here.